// @squarecloud/api/lib/types/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationFileType = exports.ApplicationStatus = exports.ApplicationTier = exports.ApplicationLanguageVersion = exports.ApplicationLanguage = exports.UserPlanName = void 0;
var UserPlanName;
(function (UserPlanName) {
    UserPlanName["Free"] = "free";
    UserPlanName["Basic"] = "basic";
    UserPlanName["Medium"] = "medium";
    UserPlanName["Advanced"] = "advanced";
    UserPlanName["Senior"] = "senior";
    UserPlanName["Deluxe"] = "deluxe";
    UserPlanName["Orion"] = "orion";
    UserPlanName["Ultimate"] = "ultimate";
})(UserPlanName = exports.UserPlanName || (exports.UserPlanName = {}));
var ApplicationLanguage;
(function (ApplicationLanguage) {
    ApplicationLanguage["JavaScript"] = "javascript";
    ApplicationLanguage["TypeScript"] = "typescript";
    ApplicationLanguage["Python"] = "python";
    ApplicationLanguage["Java"] = "java";
    ApplicationLanguage["Rust"] = "rust";
    ApplicationLanguage["Go"] = "go";
})(ApplicationLanguage = exports.ApplicationLanguage || (exports.ApplicationLanguage = {}));
var ApplicationLanguageVersion;
(function (ApplicationLanguageVersion) {
    ApplicationLanguageVersion["Recommended"] = "recommended";
    ApplicationLanguageVersion["Latest"] = "latest";
})(ApplicationLanguageVersion = exports.ApplicationLanguageVersion || (exports.ApplicationLanguageVersion = {}));
var ApplicationTier;
(function (ApplicationTier) {
    ApplicationTier["Free"] = "free";
    ApplicationTier["Paid"] = "paid";
})(ApplicationTier = exports.ApplicationTier || (exports.ApplicationTier = {}));
var ApplicationStatus;
(function (ApplicationStatus) {
    ApplicationStatus["Exited"] = "exited";
    ApplicationStatus["Created"] = "created";
    ApplicationStatus["Starting"] = "starting";
    ApplicationStatus["Restarting"] = "restarting";
    ApplicationStatus["Deleting"] = "deleting";
    ApplicationStatus["Running"] = "running";
})(ApplicationStatus = exports.ApplicationStatus || (exports.ApplicationStatus = {}));
var ApplicationFileType;
(function (ApplicationFileType) {
    ApplicationFileType["Directory"] = "directory";
    ApplicationFileType["File"] = "file";
})(ApplicationFileType = exports.ApplicationFileType || (exports.ApplicationFileType = {}));
