// @squarecloud/api/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SquareCloudAPI = void 0;
const assertions_1 = require("./assertions");
const api_1 = __importDefault(require("./managers/api"));
const application_1 = __importDefault(require("./managers/application"));
const experimental_1 = __importDefault(require("./managers/experimental"));
const user_1 = __importDefault(require("./managers/user"));
class SquareCloudAPI {
    static apiInfo = {
        latestVersion: 'v2',
        baseUrl: 'https://api.squarecloud.app/',
    };
    apiManager;
    /** Use experimental features */
    experimental;
    /** The applications manager */
    applications;
    /** The users manager */
    users;
    /**
     * Creates an API instance
     *
     * @param apiKey - Your API Token (generate at [Square Cloud Dashboard](https://squarecloud.app/dashboard))
     * @param options.experimental - Whether to enable experimental features
     */
    constructor(apiKey, options) {
        (0, assertions_1.validateString)(apiKey, 'API_KEY');
        this.apiManager = new api_1.default(apiKey);
        this.experimental = options?.experimental
            ? new experimental_1.default(this.apiManager)
            : undefined;
        this.applications = new application_1.default(this.apiManager);
        this.users = new user_1.default(this.apiManager);
    }
}
exports.SquareCloudAPI = SquareCloudAPI;
