// @squarecloud/api/lib/managers/api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const error_1 = __importDefault(require("../structures/error"));
class APIManager {
    apiKey;
    baseUrl = 'https://api.squarecloud.app';
    constructor(apiKey) {
        this.apiKey = apiKey;
    }
    user(userId) {
        return this.fetch('user' + (userId ? `/${userId}` : ''));
    }
    application(path, appId, options) {
        if (typeof options === 'string') {
            options = {
                method: options,
            };
        }
        return this.fetch(`apps/${appId}/${path}`, options);
    }
    async fetch(path, options = {}, version = 'v2', rootPath) {
        options = {
            ...options,
            method: options.method || 'GET',
            headers: { ...(options.headers || {}), Authorization: this.apiKey },
        };
        const res = await fetch(`${this.baseUrl}/${version}${rootPath ? `/${rootPath}` : ''}/${path}`, options).catch((err) => {
            throw new error_1.default(err.code);
        });
        const data = await res.json();
        if (!data || data.status === 'error' || !res.ok) {
            throw new error_1.default(data?.code || 'COMMON_ERROR');
        }
        return data;
    }
}
exports.default = APIManager;
