// @squarecloud/api/lib/managers/experimental.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const assertions_1 = require("../assertions");
class ExperimentalManager {
    apiManager;
    constructor(apiManager) {
        this.apiManager = apiManager;
    }
    /**
     * @experimental
     * **May have bugs.**
     * The new Square Cloud experimental AI feature.
     *
     * @param prompt.question - Short question for the AI to answer :)
     * @param prompt.context - Subject, humor or previous conversations
     */
    async askAi(prompt) {
        if (prompt.context) {
            (0, assertions_1.validateString)(prompt.context);
        }
        (0, assertions_1.validateString)(prompt.question);
        const { response } = await this.apiManager.fetch('ai', {
            method: 'POST',
            body: JSON.stringify({
                question: prompt.question,
                prompt: prompt.context,
            }),
        }, 'v1', 'experimental');
        return response;
    }
}
exports.default = ExperimentalManager;
